### PR TITLE
Improve Desktop filesharing docs (stable 2.2.0.0)

### PR DESCRIPTION
**Pull request by djs55**
This is a subset of #10455 describing current Desktop stable 2.2.0.0.

### Proposed changes

In Docker Desktop 2.2.0.0 stable the filesharing implementation was changed from Samba to a FUSE-based system much more similar to what is used on the Mac. This means that inotify now works.

There were some minor improvements to filesharing on the Mac, so I've included that too.

### Unreleased project version (optional)

This applies to released Docker Desktop 2.2.0.0 stable.

### CODE REVIEW

File: docker-for-windows/troubleshoot.md [hunk 1]
```diff
@@ -148,10 +134,9 @@ containers). Click ![whale menu](images/whale-x.png){: .inline}
  and then **Settings** > **Shared Drives** and share the drive that contains the
 Dockerfile and volume.
 
-#### Understand symlinks limitations
+#### Support for symlinks
 
-Symlinks work within and across containers. However, symlinks created outside of
-containers (for example, on the host) do not work. To learn more, see [Are
+Symlinks work within and across containers. To learn more, see [Are
 symlinks supported?](faqs.md#are-symlinks-supported) in the FAQs.
```
Comment (usha-mandya) [inline docker-for-windows/troubleshoot.md]: Guess this link should be updated to [How do symlinks work on Windows?](faqs.md#how-do-symlinks-work-on-windows)

File: docker-for-windows/index.md [hunk 2]
```diff
@@ -228,25 +228,25 @@ You can also move the disk image to a different location. If you attempt to move
...
 >
- * You can share local drives with your _containers_ but not with Docker Machine
-nodes. See the FAQ, [Can I share local drives and filesystem with my Docker Machine VMs?](faqs.md#can-i-share-local-drives-and-filesystem-with-my-docker-machine-vms).
->
+ * Windows presents a case-insensitive view of the filesystem to applications while Linux is case-sensitive. On Linux it is possible to create 2 separate files: `test` and `Test`, while on Windows these filenames would actually refer to the same underlying file. This can lead to problems where an app works correctly on a developer Windows machine (where the file contents are shared) but fails when run in Linux in production (where the file contents are distinct). To avoid this Docker Desktop insists that all shared files are accessed as their original case. Therefore if a file is created called `test`, it must be opened as `test`. Attempts to open `Test` will fail with "No such file or directory". Similarly once a file called `test` is created, attempts to create a second file called `Test` will fail.
```
Comment (usha-mandya) [inline docker-for-windows/index.md]: nit: To avoid this, Docker Desktop insists...

**Comment by usha-mandya** [comment]
Thank you @djs55. Suggested some minor updates. Looks good otherwise.

File: docker-for-windows/troubleshoot.md (hunk 1 above)
Comment (djs55) [reply docker-for-windows/troubleshoot.md]: Oh good spot. It's a bit ironic that the link to the section on symlinks was broken :)

**Pull request merged**